src/expreplay.py

```diff
@@ -132,6 +146,16 @@ def _assign(self, pos, exp):
             self.reward[i, pos] = exp[2][i]
             self.isOver[i, pos] = exp[3][i]
 
+    def _assign_state(self, pos, obs):
+        for i in range(self.agents):
+            self.state[i, pos] = obs[i]
```

Comment:
If both `self.state` and `obs` are numpy arrays, could we avoid the for loop and make the assignment faster with something like `self.state[:, pos] = obs[:]`? Similarly for the method `_assign_effect`